stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -208,6 +214,96 @@ public void run() {
     });
   }
 
+  @ChromeDevtoolsMethod
+  public PerformSearchResponse performSearch(JsonRpcPeer peer, JSONObject params) {
+    final PerformSearchRequest request = mObjectMapper.convertValue(
+        params,
+        PerformSearchRequest.class);
+
+    // Searches are case insensitive so convert
+    // the query to lower case for simpler comparisons.
+    request.query = request.query.toLowerCase();
+
+    final Object root = this.mShadowDOM.getRootElement();
+
+    // Each search action has a unique ID so that
+    // it can be queried later.
+    final String id = UUID.randomUUID().toString();
+
+    final ArrayList<Integer> results = new ArrayList<>();
+    FindMatches(request.query, root, results);
+
+    mSearchResults.put(id, results);
+
+    final PerformSearchResponse response = new PerformSearchResponse();
+    response.searchId = id;
+    response.resultCount = results.size();
+
+    return response;
+  }
+
+  private void FindMatches(String query, Object root, ArrayList<Integer> resultIds) {
+    final ElementInfo info = mShadowDOM.getElementInfo(root);
+
+    for (int i = 0; i < info.children.size(); i++) {
+      final Object element = info.children.get(i);
+
+      if (DoesElementMatch(query, element)) {
+        resultIds.add(mObjectIdMapper.getIdForObject(element));
+      }
+
+      FindMatches(query, element, resultIds);
+    }
+  }
+
+  private boolean DoesElementMatch(String query, Object element) {
+    final String queryParam = query;
```

Comment:
you can avoid copying into a local variable just for the sake of adding `final` ... by adding `final` to the parameter, e.g.
```
private boolean doesElementMatch(final String query, final Object element) {
```